code/modules/awaymissions/maploader/reader.dm

```diff
@@ -174,14 +175,14 @@ var/global/dmm_suite/preloader/_preloader = null
 	//first instance the /area and remove it from the members list
 	index = members.len
 	var/atom/instance
-	_preloader = new(members_attributes[index])//preloader for assigning  set variables on atom creation
+	_preloader.setup(members_attributes[index])//preloader for assigning  set variables on atom creation

```

Comment:
Checking for any attribute can be done here (and below) instead of inside ```preloader/setup()```, if you want to spare the function call.